toolchain/check/testdata/class/fail_adapt_modifiers.carbon

```diff
@@ -0,0 +1,110 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+class B {}
+
+class C1 {
+  // CHECK:STDERR: fail_adapt_modifiers.carbon:[[@LINE+4]]:3: ERROR: `private` not allowed on `adapt` declaration.
+  // CHECK:STDERR:   private adapt B;
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR:
+  private adapt B;
+}
+
+class C2 {
+  // CHECK:STDERR: fail_adapt_modifiers.carbon:[[@LINE+4]]:3: ERROR: `abstract` not allowed on `adapt` declaration.
+  // CHECK:STDERR:   abstract adapt B;
+  // CHECK:STDERR:   ^~~~~~~~
+  // CHECK:STDERR:
+  abstract adapt B;
+}
+
+class C3 {
+  // CHECK:STDERR: fail_adapt_modifiers.carbon:[[@LINE+4]]:3: ERROR: `default` not allowed on `adapt` declaration.
+  // CHECK:STDERR:   default adapt B;
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR:
+  default adapt B;
+}
+
+class C4 {
+  // CHECK:STDERR: fail_adapt_modifiers.carbon:[[@LINE+6]]:10: ERROR: `extend` repeated on declaration.
+  // CHECK:STDERR:   extend extend adapt B;
+  // CHECK:STDERR:          ^~~~~~
+  // CHECK:STDERR: fail_adapt_modifiers.carbon:[[@LINE+3]]:3: `extend` previously appeared here.
+  // CHECK:STDERR:   extend extend adapt B;
+  // CHECK:STDERR:   ^~~~~~
+  extend extend adapt B;
```

Comment:
extend base adapt should probably be tested due to the special-ness of base. :)

Reply:
Done.